File: backend/src/dataclasses/playlist.py
from pydantic import BaseModel
from typing import List

from src.dataclasses.followers import Followers

from src.dataclasses.external_urls import ExternalUrls
from src.dataclasses.image import Image
from src.dataclasses.playlist_tracks import PlaylistTracks
from src.dataclasses.user import User


class Playlist(BaseModel):
    collaborative: bool
    description: str
    external_urls: ExternalUrls
    followers: Followers
    href: str
    id: str
    images: List[Image]
    name: str
    owner: User
    public: bool
    snapshot_id: str
    tracks: PlaylistTracks
    type: str
    uri: str

File: react-native-frontend/PlaylistManager/interfaces/Playlist.ts
import { User } from "./User";
import { Track } from "./Track";

export interface Playlist {
  collaborative: false;
  name: string;
  description: string;
  external_urls: {
    spotify: string;
  };
  href: string;
  id: string;
  image_url?: string;
  owner: User;
  public: false;
  snapshot_id: string;
  tracks: Track[];
  type: string;
  uri: string;
}
